source/common/network/connection_impl.cc

```diff
@@ -656,8 +656,6 @@ void ConnectionImpl::onWriteReady() {
     socklen_t error_size = sizeof(error);
     Api::SysCallIntResult result =
         socket_->getSocketOption(SOL_SOCKET, SO_ERROR, &error, &error_size);
-    RELEASE_ASSERT(result.rc_ == 0,
-                   fmt::format("Failed to connect: {}", errorDetails(result.errno_)));
 
     if (error == 0) {
```

Comment:
undefined behavior: error is not inialized if result.rc_ != 0

Reply:
I see, I was under the impression `error` would be set, but it's actually two separate things. I'll handle getSocketOption return error with an `ENVOY_BUG` (since this is unexpected) and close socket,  and `error != 0` additionally with close socket.

Checking my understanding. `getSocketOption` can fail for e.g. if the fd is invalid, or we don't have enough resources. If we succeed, the socket may *still* have a pending error returned with `error` that happened between socket calls. Maybe like connection refused?